rocAL/source/decoders/image/rocjpeg_decoder.cpp

```diff
@@ -0,0 +1,386 @@
+            break;
+        case ROCJPEG_CSS_400:
+            chroma_sub_sampling = "YUV 4:0:0";
+            break;
+        case ROCJPEG_CSS_UNKNOWN:
+            chroma_sub_sampling = "UNKNOWN";
+            break;
+        default:
+            chroma_sub_sampling = "";
+            break;
+    }
+}
+
+void HWRocJpegDecoder::initialize(int device_id, unsigned batch_size) {
+    int num_devices;
+    CHECK_HIP(hipGetDeviceCount(&num_devices));
+    if (num_devices < 1) {
+        std::cerr << "ERROR: didn't find any GPU!" << std::endl;
+        return;
+    }
+    if (device_id >= num_devices) {
+        std::cerr << "ERROR: the requested device_id is not found!" << std::endl;
+        return;
+    }
+    CHECK_HIP(hipSetDevice(device_id));
+    RocJpegBackend rocjpeg_backend = ROCJPEG_BACKEND_HARDWARE;
+    // Create stream and handle
+    CHECK_ROCJPEG(rocJpegCreate(rocjpeg_backend, device_id, &_rocjpeg_handle));
+    _rocjpeg_streams.resize(batch_size);
+    for (unsigned i = 0; i < batch_size; i++) {
+        CHECK_ROCJPEG(rocJpegStreamCreate(&_rocjpeg_streams[i]));
+    }
+
+    _device_id = device_id;
+    _batch_size = batch_size;
+    _output_images.resize(_batch_size);
+    _src_hstride.resize(_batch_size);
+    _src_img_offset.resize(_batch_size);
+
+    // Allocate mem for width and height arrays for src and dst
+    if (!_dev_src_width) CHECK_HIP(hipMalloc((void **)&_dev_src_width, _batch_size * sizeof(size_t)));
+    if (!_dev_src_height) CHECK_HIP(hipMalloc((void **)&_dev_src_height, _batch_size * sizeof(size_t)));
+    if (!_dev_dst_width) CHECK_HIP(hipMalloc((void **)&_dev_dst_width, _batch_size * sizeof(size_t)));
+    if (!_dev_dst_height) CHECK_HIP(hipMalloc((void **)&_dev_dst_height, _batch_size * sizeof(size_t)));
+    if (!_dev_src_hstride) CHECK_HIP(hipMalloc((void **)&_dev_src_hstride, _batch_size * sizeof(size_t)));
+    if (!_dev_src_img_offset) CHECK_HIP(hipMalloc((void **)&_dev_src_img_offset, _batch_size * sizeof(size_t)));
+
+}
+
+// Obtains the decode info of the image, and modifies width and height based on the max decode params after scaling
+Decoder::Status HWRocJpegDecoder::decode_info(unsigned char *input_buffer, size_t input_size, int *width, int *height, int *actual_width, 
+                                              int *actual_height, int max_decoded_width, int max_decoded_height, Decoder::ColorFormat desired_decoded_color_format, int index) {
+    RocJpegChromaSubsampling subsampling;
+    uint8_t num_components;
+    uint32_t widths[4] = {};
+    uint32_t heights[4] = {};
+    uint32_t channels_size = 0;
+    uint32_t channel_sizes[4] = {};
+
+    uint32_t max_widths[4] = {static_cast<uint32_t>(max_decoded_width), 0, 0, 0};
+    uint32_t max_heights[4] = {static_cast<uint32_t>(max_decoded_height), 0, 0, 0};
+
+    switch(desired_decoded_color_format) {
+        case Decoder::ColorFormat::GRAY:
+            _decode_params.output_format = ROCJPEG_OUTPUT_Y;
```

Comment:
please modify the `_decode_params.output_format` to _`decode_params[index].output_format`

Reply:
Done